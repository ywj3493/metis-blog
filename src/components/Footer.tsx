"use client";

export default function Footer() {
  return (
    <footer className="bg-black text-white text-sm text-center">Footer</footer>
  );
}
