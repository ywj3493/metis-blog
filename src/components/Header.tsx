import Image from "next/image";
import Link from "next/link";

type Menu = {
  url: string;
  name: string;
};

const menus: Menu[] = [
  {
    url: "/about",
    name: "소개",
  },
  {
    url: "/contact",
    name: "연락",
  },
  {
    url: "/posts",
    name: "게시글",
  },
];

export default function Header() {
  return (
    <header className="bg-white flex justify-between items-center p-16 box-border">
      <Link className="clickable flex items-center gap-4" href="/">
        <Image src={"/mascot.png"} height={48} width={48} alt={"icon"} />
        <h1 className="text-2xl font-bold">{"메티의 Notion"}</h1>
      </Link>
      <nav className="flex gap-16 clickable">
        {menus.map(({ name, url }) => (
          <Link key={name} href={url}>
            {name}
          </Link>
        ))}
      </nav>
    </header>
  );
}
