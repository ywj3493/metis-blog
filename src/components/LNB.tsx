"use client";

import { Dispatch, SetStateAction } from "react";
import TooltipWrapper from "./TooltipWrapper";
import { Tag } from "@/adapters/posts";
import TagChip from "./posts/Tag";

type LNBProps = {
  tags: Tag[];
  selectedTags: Set<string>;
  setSelectedTags: Dispatch<SetStateAction<Set<string>>>;
};

export default function LNB({ tags, selectedTags, setSelectedTags }: LNBProps) {
  const handleTagClick = (id: string) => {
    if (selectedTags.has(id)) {
      setSelectedTags((prev) => {
        const newSet = new Set(prev);
        newSet.delete(id);
        return newSet;
      });
    } else {
      setSelectedTags((prev) => {
        const newSet = new Set(prev);
        newSet.add(id);
        return newSet;
      });
    }
  };

  return (
    <div className="flex flex-col w-200 gap-4 items-center sticky top-0 h-min mt-8">
      <TooltipWrapper message="태그를 여러개 선택할 수 있습니다. or 조건으로 검색됩니다.">
        <div className="border-b-2 border-blue-200 px-8 cursor-default">
          Tags
        </div>
      </TooltipWrapper>
      {tags.map(({ id, name, color }) => (
        <TagChip
          key={id}
          id={id}
          name={name}
          color={color}
          notSelected={!selectedTags.has(id)}
          onClick={handleTagClick}
        />
      ))}
    </div>
  );
}
