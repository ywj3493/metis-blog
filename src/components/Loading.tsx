export function LoadingSpinner() {
  return (
    <div className="flex justify-center items-center">
      <div className="animate-spin w-32 h-32 border-4 border-gray-200 border-b-blue-300 rounded-full inline-block box-border "></div>
    </div>
  );
}

export function LoadingDot() {
  return (
    <div className="animate-pulse rounded-full w-24 h-24 bg-gray-300"></div>
  );
}
