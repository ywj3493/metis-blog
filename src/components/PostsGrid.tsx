import { DatabaseObjectResponse } from "@notionhq/client/build/src/api-endpoints";
import PostCard from "./posts/PostCard";

type PostsGridProps = {
  posts: DatabaseObjectResponse[];
};

export default function PostsGrid({ posts }: PostsGridProps) {
  return (
    <ul className="grid gap-16 grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4">
      {posts.map((post) => (
        <PostCard key={post.id} data={post} />
      ))}
    </ul>
  );
}
