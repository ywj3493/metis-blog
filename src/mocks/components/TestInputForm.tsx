"use client";

import axios from "axios";
import { useEffect, useRef, useState } from "react";
import type { TestEmployee, TestEmployeeRequest } from "../api/test";

interface TestEmployeeItemProps extends TestEmployee {
  onClickDelete: () => void;
}

function TestEmployeeItem({
  name,
  age,
  empNumber,
  onClickDelete,
}: TestEmployeeItemProps) {
  return (
    <div className="grid grid-cols-4 h-48 rounded-5 border-blue-200 items-center bg-white p-8">
      <div className="font-r-16 text-black">{`사번: ${empNumber}`}</div>
      <div className="font-r-16 text-black">{`이름: ${name}`}</div>
      <div className="font-r-16 text-black">{`나이: ${age}`}</div>
      <button
        type="button"
        className="bg-red-600 text-white p-5 rounded-5"
        onClick={onClickDelete}
      >
        delete
      </button>
    </div>
  );
}

const getTestEmployee = async () => {
  const { data } = await axios.get<TestEmployee[]>("/emplyee");

  return data;
};

const postTestEmployee = async (request: TestEmployeeRequest) => {
  const { data } = await axios.post<TestEmployeeRequest>("/employee", request);

  return data;
};

const deleteTestEmplyee = async (empNumber: string) => {
  const { data } = await axios.delete(`/employee/${empNumber}`);

  return data;
};

export function TestInputForm() {
  const nameInputRef = useRef<HTMLInputElement>(null);
  const ageInputRef = useRef<HTMLInputElement>(null);

  const [employees, setEmployees] = useState<TestEmployee[]>([]);

  const [postTrigger, setPostTrigger] = useState(false);

  const handlePost = async () => {
    const name = nameInputRef.current?.value;
    const age = ageInputRef.current?.value;
    if (!!name && !!age) {
      await postTestEmployee({
        name: name,
        age: Number.parseInt(age),
      });
    }
    setPostTrigger((prev) => !prev);
  };

  const handleDelete = async (empNumber: string) => {
    await deleteTestEmplyee(empNumber);
    setPostTrigger((prev) => !prev);
  };

  useEffect(() => {
    const fetch = async () => {
      const data = await getTestEmployee();
      setEmployees(data);
    };
    try {
      fetch();
    } catch (e) {}
  }, []);

  return (
    <>
      <div className="flex flex-col gap-10">
        {employees?.map(({ name, age, empNumber }) => (
          <TestEmployeeItem
            key={empNumber}
            name={name}
            age={age}
            empNumber={empNumber}
            onClickDelete={() => handleDelete(empNumber)}
          />
        ))}
      </div>
      <div className="flex flex-col gap-20">
        <div className="font-r-16 text-black">Name</div>
        <input ref={nameInputRef} placeholder="name" />
        <div className="font-r-16 text-black">Age</div>
        <input ref={ageInputRef} type="number" placeholder="age" />
        <button
          type="button"
          className="bg-cyan-700 text-white p-5 rounded-5"
          onClick={handlePost}
        >
          POST
        </button>
      </div>
    </>
  );
}
