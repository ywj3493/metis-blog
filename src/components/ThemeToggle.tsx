"use client";

import { useTheme } from "next-themes";
import { useEffect, useState } from "react";
import { MdDarkMode as DarkModeIcon } from "react-icons/md";
import { MdOutlineDarkMode as LightModeIcon } from "react-icons/md";
import { LoadingDot } from "./Loading";

export default function ThemeToggle() {
  const [mounted, setMounted] = useState(false);
  const { theme, setTheme } = useTheme();

  useEffect(() => setMounted(true), []);

  return (
    <div className="aspect-square w-24 h-24">
      {mounted ? (
        theme === "light" ? (
          <DarkModeIcon
            onClick={() => setTheme("dark")}
            className="aspect-square w-full h-full"
          />
        ) : (
          <LightModeIcon
            onClick={() => setTheme("light")}
            className="aspect-square w-full h-full"
          />
        )
      ) : (
        <LoadingDot />
      )}
    </div>
  );
}
