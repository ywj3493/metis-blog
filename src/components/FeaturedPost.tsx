import { Post } from "@/adapters/posts";
import PostsGrid from "./PostsGrid";
import { getNotionPosts } from "@/services/_external";

export default async function FeaturedPosts() {
  const dataList = await getNotionPosts();
  const posts = dataList.map(Post.create);
  return (
    <section className="p-24 w-full">
      <h2 className="text-2xl font-bold my-8 px-8">Featured Posts</h2>
      <PostsGrid posts={posts} />
    </section>
  );
}
