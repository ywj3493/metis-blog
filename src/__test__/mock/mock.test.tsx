import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { describe, expect, it } from "vitest";
import { TestInputForm } from "@/mocks/components/TestInputForm";

describe("MockingTest in client", () => {
  it("should render correctly", async () => {
    const testInputForm = render(<TestInputForm />);
    expect(testInputForm).toMatchSnapshot();
  });

  it("should input correctly", async () => {
    render(<TestInputForm />);
    const nameInput = screen.getByPlaceholderText("name");
    const ageInput = screen.getByPlaceholderText("age");
    const postButton = screen.getByText("POST");

    await userEvent.type(nameInput, "김영수");
    await userEvent.type(ageInput, "2");
    await userEvent.click(postButton);

    const item = screen.getByText("사번: 1");
    expect(item).toBeInTheDocument();

    const deleteButton = screen.getByText("delete");

    await userEvent.click(deleteButton);
    expect(item).not.toBeInTheDocument();
  });
});
