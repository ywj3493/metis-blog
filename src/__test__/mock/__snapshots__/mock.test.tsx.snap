// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MockingTest > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1h1oj6e-ListWrapper ep7bpre5"
      />
      <div
        class="css-11qwdcn-FormWrapper ep7bpre6"
      >
        <div
          class="css-va76tc-Title ep7bpre2"
        >
          Name
        </div>
        <input
          placeholder="name"
        />
        <div
          class="css-va76tc-Title ep7bpre2"
        >
          Age
        </div>
        <input
          placeholder="age"
          type="number"
        />
        <button
          class="css-1acsazi-SubmitButton ep7bpre3"
        >
          POST
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1h1oj6e-ListWrapper ep7bpre5"
    />
    <div
      class="css-11qwdcn-FormWrapper ep7bpre6"
    >
      <div
        class="css-va76tc-Title ep7bpre2"
      >
        Name
      </div>
      <input
        placeholder="name"
      />
      <div
        class="css-va76tc-Title ep7bpre2"
      >
        Age
      </div>
      <input
        placeholder="age"
        type="number"
      />
      <button
        class="css-1acsazi-SubmitButton ep7bpre3"
      >
        POST
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`MockingTest in client > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col gap-10"
      />
      <div
        class="flex flex-col gap-20"
      >
        <div
          class="font-r-16 text-black"
        >
          Name
        </div>
        <input
          placeholder="name"
        />
        <div
          class="font-r-16 text-black"
        >
          Age
        </div>
        <input
          placeholder="age"
          type="number"
        />
        <button
          class="bg-cyan-700 text-white p-5 rounded-5"
        >
          POST
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col gap-10"
    />
    <div
      class="flex flex-col gap-20"
    >
      <div
        class="font-r-16 text-black"
      >
        Name
      </div>
      <input
        placeholder="name"
      />
      <div
        class="font-r-16 text-black"
      >
        Age
      </div>
      <input
        placeholder="age"
        type="number"
      />
      <button
        class="bg-cyan-700 text-white p-5 rounded-5"
      >
        POST
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
